### PR TITLE
EPMRPP-42197 || Item ids holding impl

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/listeners/ListenerParameters.java [hunk 1]
```diff
@@ -335,6 +347,7 @@ public String toString() {
 		sb.append(", rerunOf='").append(rerunOf).append('\'');
 		sb.append(", asyncReporting=").append(asyncReporting);
 		sb.append(", ioPoolSize=").append(ioPoolSize);
+    sb.append(", callbackReportingEnabled=").append(callbackReportingEnabled);
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/listeners/ListenerParameters.java]: Formatting

File: src/main/java/com/epam/reportportal/service/tree/TestItemTree.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+import com.epam.ta.reportportal.ws.model.OperationCompletionRS;
+import io.reactivex.Maybe;
+import io.reactivex.annotations.Nullable;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/service/tree/TestItemTree.java]: Add some documentation

File: src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Function;
+import io.reactivex.schedulers.Schedulers;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java]: Add some documentation

File: src/main/java/com/epam/reportportal/listeners/ListenerParameters.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/listeners/ListenerParameters.java]: fixed

File: src/main/java/com/epam/reportportal/service/tree/TestItemTree.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/service/tree/TestItemTree.java]: added

File: src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java]: added

File: src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+ * This class provides methods for sending requests to the Report Portal instance, using {@link ReportPortalClient}
+ * and {@link com.epam.reportportal.service.tree.TestItemTree.TestItemLeaf}.
+ * Provided requests:
+ * - start test item
+ * - finish test item
+ * - send log
+ *
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public class ItemTreeReporter {
```
Comment (HardNorth) [inline src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java]: I have no idea how I suppose to use this class on practice. Could you provide any examples?

File: src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+	 * @param launchId           Launch id
+	 * @param testItemLeaf       {@link com.epam.reportportal.service.tree.TestItemTree.TestItemLeaf}
+	 * @return {@link Maybe} containing item id
+	 */
+	public static Maybe<String> startItem(ReportPortalClient reportPortalClient, final StartTestItemRQ startTestItemRQ,
+			final Maybe<String> launchId, final TestItemTree.TestItemLeaf testItemLeaf) {
+		final Maybe<String> parent = testItemLeaf.getParentId();
+		if (parent != null && launchId != null) {
+			Maybe<String> itemId = sendStartItemRequest(reportPortalClient, launchId, parent, startTestItemRQ);
+			itemId.subscribeOn(Schedulers.computation()).subscribe();
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java]: updated

File: src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java (hunk 4 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java]: repository wiki updated with description

File: src/main/java/com/epam/reportportal/listeners/ListenerParameters.java [hunk 6]
```diff
@@ -42,6 +41,7 @@
 	private static final boolean DEFAULT_CONVERT_IMAGE = false;
 	private static final boolean DEFAULT_RETURN = false;
 	private static final boolean DEFAULT_ASYNC_REPORTING = false;
+	private static final boolean DEFAULT_CALLBACK_REPORTING_ENABLED = false;
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/listeners/ListenerParameters.java]: Callback reporting is required for special cases so there is no need to enable it by default and consume additional resources

**Pull request merged**